cdk/domino_cdk/provisioners/eks/eks_cluster.py

```diff
@@ -37,6 +39,16 @@ def provision(
             version=eks_version,
             default_capacity=0,
             security_group=eks_sg,
+            secrets_encryption_key=Key.from_key_arn(
+                self.scope, "secrets_encryption_key_arn", secrets_encryption_key_arn
+            )
+            if secrets_encryption_key_arn
+            else Key(
+                self.scope,
+                f"{name}-kubernetes-secrets-envelope-key",
+                alias=f"{name}-kubernetes-secrets-envelope-key",
+                removal_policy=cdk.RemovalPolicy.DESTROY,
+            ),
```

Comment:
Can we also add `enableKeyRotation=true`?

Reply:
Need to try. The doc doesn't say we cannot.

Reply:
This works at least during deployment.

Reply:
Thanks! It comes in the CIS compliance scans.

Reply:
Just a personal preference thing, but this ternary block is big enough that I'd be inclined to evaluate it before the call to `eks.Cluster()`